### DISCUSSION TITLE
Network stack does not respond to ARP packets or any other connection

### DISCUSSION THREAD

**Discussion by omersiar** [rsta2/circle#493] [Q&A] [answered]
Hello,

Thank you for making it possible for us to tinker Raspberries.

I was trying to add networking capabilities to [MiniDexed](https://github.com/probonopd/MiniDexed/wiki/Development) project, and implementing the [mt32's network](https://github.com/dwhinham/mt32-pi/wiki/Networking) features.

It seems everything should work, although even Raspberry Pi 3A+ can connect to the Accees Point got an IP, code successfully binds UDP ports I can call CNetSubSystem::Get(); to figure out the DHCP IP. But when I try to connect to bound UDP port Raspberry simply does not respond to any network packages starting from ARP requests.

It seems MiniDexed run on some loop that does not yield to net subsystem to process packages.  I have included single routine to [what seems to check for arrived packages](https://github.com/dwhinham/mt32-pi/blob/ee645df6e38508e16e6a265fe89d13ce5c815b65/src/mt32pi.cpp#L426) but that does not help.

Diff if you want to take a look.
https://github.com/probonopd/MiniDexed/compare/2024-01-16...omersiar:MiniDexed:network

@probonopd tagging you here because this is where I got stuck now, tried to implement RTP receiver as a virtual MIDI device.

The question is how can I troubleshoot network connectivity?  What to call for network related updates like arriving packages from other loops?

**Comment by rsta2** [comment, marked as answer]
Do you call CScheduler::Get()->Yield() somewhere in the main loop on core 0? 
This is necessary, because Circle has a cooperative scheduler, which runs on 
core 0 like the networking code and all tasks must block, sleep or call yield 
from time to time to allow the other tasks to run.

**Reply by omersiar** [reply]
That was it.  Thank you Sir/Madam.

**Comment by omersiar** [comment]
Thanks for the explanation, let me do that.
